// db/table-definitions.js
module.exports = {
	user: {
		model: 'user',
		tableName: 'testing-user',
		keySchema: [
      { AttributeName: 'email', KeyType: 'HASH' },
    ],
    attributeDefinitions: [
      { AttributeName: 'email', AttributeType: 'S' },
    ],
    provisionedThroughput: {
      ReadCapacityUnits: 1,
      WriteCapacityUnits: 1,
    }
	},
	token: {
		model: 'token',
		tableName: 'testing-token-1',
		keySchema: [
      { AttributeName: 'tokenHash', KeyType: 'HASH' }
    ],
    attributeDefinitions: [
      { AttributeName: 'tokenHash', AttributeType: 'S' },
      { AttributeName: 'email', AttributeType: 'S' }
    ],
    GlobalSecondaryIndexes: [
      {
        IndexName: 'email-index',
        KeySchema: [
          {
            AttributeName: 'email',
            KeyType: 'HASH'
          }
        ],
        Projection: {
          ProjectionType: 'ALL'
        },
        ProvisionedThroughput: {
          ReadCapacityUnits: 1,
          WriteCapacityUnits: 1
        }
      }
    ],    
    provisionedThroughput: {
      ReadCapacityUnits: 1,
      WriteCapacityUnits: 1,
    }
	},
  content: {
    model: 'content',
    tableName: 'testing-content',
    keySchema: [
      { AttributeName: 'id', KeyType: 'HASH' }
    ],
    attributeDefinitions: [
      { AttributeName: 'id', AttributeType: 'S' },
      { AttributeName: 'email', AttributeType: 'S' }
    ],
    provisionedThroughput: {
      ReadCapacityUnits: 1,
      WriteCapacityUnits: 1
    },
    GlobalSecondaryIndexes: [
      {
        IndexName: 'email-index',
        KeySchema: [
          {
            AttributeName: 'email',
            KeyType: 'HASH'
          }
        ],
        Projection: {
          ProjectionType: 'ALL'
        },
        ProvisionedThroughput: {
          ReadCapacityUnits: 1,
          WriteCapacityUnits: 1
        }
      }
    ],    
    provisionedThroughput: {
      ReadCapacityUnits: 1,
      WriteCapacityUnits: 1,
    }    
  }
}
